src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -29,26 +29,32 @@ public static (List<Error> errors, List<TableOfContentsItem> items, JObject meta
             return (errors, items, inputModel?.Metadata, fileMonikers);
         }
 
-        private static (List<Error> errors, TableOfContentsInputModel tocModel) LoadTocModel(Context context, Document file)
+        private static (List<Error> errors, TableOfContentsInputModel tocModel) LoadTocModel(Context context, Document file, string content = null)
         {
             var filePath = file.FilePath;
+
+            if (file.IsFromHistory)
+            {
+                Debug.Assert(!string.IsNullOrEmpty(content));
+            }
+
             if (filePath.EndsWith(".yml", PathUtility.PathComparison))
             {
-                var (errors, tocToken) = YamlUtility.Deserialize(file, context);
+                var (errors, tocToken) = !file.IsFromHistory ? YamlUtility.Deserialize(file, context) : YamlUtility.Deserialize(content);
```

Comment:
nit: Instead of do the switch based on `IsFromHistory`, how about do the switch based on `content != null` ?

Reply:
I prefer to use the IsFromHistory which is more clear to me why we need do that